### PR TITLE
fix #1872 Avoids classloader leak due to backtrace field in static throwables 

### LINKED ISSUE #1872: Classloader leak with Throwable in static fields
In dynamic classloader environments, the use of Throwable in static fields may cause a classloader leak in the backtrace field. This field is set in runtime at native code with an array that are instances of Class. This may cause a Classloader which should be disposed to be retained.

In other [projects](https://github.com/eclipse/jetty.project/issues/868), an error without a writable stacktrace was used for this cases. 

I've opened a PR for this:
https://github.com/reactor/reactor-core/pull/1871

Let me know.

Thanks!

### PR DESCRIPTION
**Pull request by simonbasle**
Supersedes #1871 
cc @fsgonz

### CODE REVIEW

**Comment by fsgonz** [comment]
@simonbasle, thank you very much for the review and taking account of this! May I help you in any way?

**Comment by simonbasle** [comment]
@fsgonz no need to apologize, I can't expect everybody to be answering right away. In other circumstances I would have waited for your reply a bit more, but time was pressing if I wanted to include this in tomorrow's release. What you can do however is to validate that the classloader leak happens with `3.2.11.RELEASE` and doesn't with `3.2.12.BUILD-SNAPSHOT`, once this PR gets merged.

**Comment by fsgonz** [comment]
@simonbasle, sure! I'll let you know. Thanks

File: reactor-core/src/main/java/reactor/core/Exceptions.java [hunk 1]
```diff
@@ -586,11 +585,56 @@ public synchronized Throwable fillInStackTrace() {
...
+		}
+	}
+
+	/**
+	 * A general-purpose {@link Throwable} that is suitable for usage as a static final
+	 * field. It avoids {@link ClassLoader}-related leaks by bypassing stacktrace filling.
+	 * Exception {{@link Exception#addSuppressed(Throwable)} suppression} is also disabled.
+	 */
+	//see https://github.com/reactor/reactor-core/pull/1872
+	static final class StaticThrowable extends Error {
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/Exceptions.java]: `StaticThrowable` does not override `fillInStackTrace`

Comment (fsgonz) [reply reactor-core/src/main/java/reactor/core/Exceptions.java]: The constructor writableStackTrace is used for that case. Is you use that, fillStackTrace is not used.

Comment (bsideup) [reply reactor-core/src/main/java/reactor/core/Exceptions.java]: TIL! 👍

**Comment by simonbasle** [comment]
damn, of course I opened and merged this against master 🤦‍♂

**Pull request merged**